### DISCUSSION TITLE
SystemError: type Boost.Python.enum has the Py_TPFLAGS_HAVE_GC flag but has no traverse function

### DISCUSSION THREAD

**Discussion by vagaiprem** [carla-simulator/carla#6704] [General]
When i open the terminal and run the python script i will get this error . 

I am installed in windows , Conda - environment :  

plz help me :

Traceback (most recent call last):
  File "C:\prem\software\CARLA_0.9.14\WindowsNoEditor\PythonAPI\examples\manual_control.py", line 83, in <module>
    import carla
  File "C:\Users\Admin\anaconda3\envs\Carla_1\Lib\site-packages\carla\__init__.py", line 8, in <module>
    from .libcarla import *
SystemError: type Boost.Python.enum has the Py_TPFLAGS_HAVE_GC flag but has no traverse function

Plz help me anyone

**Comment by BuRaGaZz** [comment]
Hye i have the same issue too
Windows 11
Run in x64 Native Tools Command Prompt for VS 2019

try to run .py in carla pythonAPI examples

C:\carla\PythonAPI\examples>py -3.11 automatic_control.py
pygame 2.5.1 (SDL 2.28.2, Python 3.11.4)
Hello from the pygame community. https://www.pygame.org/contribute.html
Traceback (most recent call last):
File "C:\carla\PythonAPI\examples\automatic_control.py", line 58, in
import carla
File "C:\Users\User\AppData\Roaming\Python\Python311\site-packages\carla_init_.py", line 8, in
from .libcarla import *
SystemError: type Boost.Python.enum has the Py_TPFLAGS_HAVE_GC flag but has no traverse function

**Comment by rrivera5611** [comment]
Any updates with this?

**Comment by vicdenal** [comment]
Try with Python 3.8. I downgraded the Python version to 3.8 - cause it was the one that worked fine with Carla 0.9.11 and after running make clean and building again, it is running OK now.

**Comment by chrisszhang** [comment]
Try to open  Util/BuildTools/Setup.sh  find  "BOOST_VERSION=1.80.0" change it  to "BOOST_VERSION=1.81.0" 
build  PythonAPI again,  it is working fine with Python3.12

**Reply by rampallishyam** [reply]
This worked for me. Thanks for sharing.
